src/groups/mqb/mqbs/mqbs_virtualstoragecatalog.h

```diff
@@ -113,11 +119,13 @@ class VirtualStorageCatalog {
     /// `rdaInfo` to the virtual storage associated with the specified
     /// `appKey`.  Note that if `appKey` is null, the message will be added
     /// to all virtual storages maintained by this instance.
-    mqbi::StorageResult::Enum put(const bmqt::MessageGUID& msgGUID,
-                                  int                      msgSize,
-                                  const bmqp::RdaInfo&     rdaInfo,
-                                  unsigned int             subScriptionId,
-                                  const mqbu::StorageKey&  appKey);
+    mqbi::StorageResult::Enum
+    put(const bmqt::MessageGUID& msgGUID,
+        int                      msgSize,
+        const bmqp::RdaInfo&     rdaInfo,
+        unsigned int             subScriptionId,
+        const mqbu::StorageKey&  appKey,
+        const OnStorageUpdateCb& putCb = OnStorageUpdateCb());
```

Comment:
Note that these `put`, `remove`, `removeAll` might also be used in the `RelayQueueEngine`, but we don't track these queue depth metrics there.